### PR TITLE
Remove use of deprecated `core-utils` methods

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This PR removes use of deprecated methods, or adds notices not to use deprecated methods. 

Blocked by #364

### CODE REVIEW

**Review by daniel-heppner-ibigroup — APPROVED** [review]

File: packages/core-utils/src/itinerary.js [hunk 1]
```diff
@@ -159,51 +181,6 @@ export function getMapColor(mode) {
...
-  return `${getStepDirection(step)} ${conjunction} ${step.streetName}`;
-}
-
-export function getStepStreetName(step) {
-  if (step.streetName === "road") return "Unnamed Road";
-  if (step.streetName === "path") return "Unnamed Path";
-  return step.streetName;
-}
-
 export function toSentenceCase(str) {
```
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/itinerary.js:184]: This one might need to be deprecated.

File: packages/core-utils/src/profile.js [hunk 2]
```diff
@@ -27,8 +27,8 @@ export function filterProfileOptions(response) {
   return response;
 }
 
-function locationString(str, defaultStr) {
-  return str ? str.split(",")[0] : defaultStr;
+function locationString(str) {
+  return str ? str.split(",")[0] : null;
```
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/profile.js:31]: `str?.split...`

File: packages/core-utils/src/query-params.js [hunk 3]
```diff
@@ -1,3 +1,7 @@
+// TODO: Remove this entire file, as it is deprecated in favor of i18n-queryParams within
```
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/query-params.js:1]: Why don't we move the deprecated contents to `deprecated.js`?

File: packages/core-utils/src/time.js [hunk 4]
```diff
@@ -92,8 +97,8 @@ export function formatDuration(seconds) {
  * @param {number} seconds duration in seconds
  * @returns {string} formatted text representation
```
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/time.js:98]: Add new param to JSdoc.

File: packages/core-utils/src/itinerary.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply packages/core-utils/src/itinerary.js:184]: There are a bunch of otp-rr keys that rely on this method to be generated. We could go through and replace all those keys, but we'd still need some form of string recapitalization/normalization. What do you think? The only other use from within otp-ui is in a deprecated method

File: packages/core-utils/src/query-params.js (hunk 3 above)
Comment (miles-grant-ibigroup) [reply packages/core-utils/src/query-params.js:1]: There's an upcoming PR that just deletes this file so I didn't see the value in moving everything over

File: packages/core-utils/src/deprecated.js [hunk 5]
```diff
@@ -0,0 +1,316 @@
+import moment from "moment";
+
+// Need to avoid cyclic dependency resolution
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/core-utils/src/deprecated.js:3]: Added in 607cb0f2263907b88396ccd78574129c94d0b1a2

File: packages/core-utils/src/profile.js (hunk 2 above)
Comment (miles-grant-ibigroup) [reply packages/core-utils/src/profile.js:31]: Refactored in 607cb0f2263907b88396ccd78574129c94d0b1a2

File: packages/core-utils/src/time.js (hunk 4 above)
Comment (miles-grant-ibigroup) [reply packages/core-utils/src/time.js:98]: Added in 607cb0f2263907b88396ccd78574129c94d0b1a2

**Pull request merged**